src/main/scala/scalaz/stream/async/mutable/Queue.scala

```diff
@@ -45,10 +45,24 @@ trait Queue[A] {
    *
    * Please use `Topic` instead of `Queue` when all subscribers
    * need to see each value enqueued.
+   *
+   * This process is equivalent to `dequeueBatch(1)`.
    */
   def dequeue: Process[Task, A]
 
   /**
+   * Provides a process that dequeues in chunks.  Whenever *n* elements
+   * are available in the queue, `min(n, limit)` elements will be dequeud
+   * and produced as a single `Seq`.  Note that this naturally does not
+   * *guarantee* that `limit` items are returned in every chunk.  If only
+   * one element is available, that one element will be returned in its own
+   * sequence.  This method basically just allows a consumer to "catch up"
+   * to a rapidly filling queue in the case where some sort of batching logic
+   * is applicable.
+   */
+  def dequeueBatch(limit: Int = 0): Process[Task, Seq[A]]
```

Comment:
So, I'm not a fan of sentinel values either.  However, they *are* sort of the standard throughout the library wherever we have these sorts of bounds.  See, `njoin`, `boundedQueue` and more.  Given that we don't have a `Nat` type in Scala, I need to handle negative/zero values in any case (just as these other functions do).  I *could* through an `IllegalArgumentException`, but I'm not a fan of mixing eager exceptional error handling with lazy on-heap schemes.  It just tends to get confusing.

Since this *is* the convention throughout the library for bounds, I'm in favor of keeping it the way I implemented and changing everything over to a better scheme down the road.  Inconsistency in either direction seems like a bad idea.